datastore/gcloud/aio/datastore/value.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any
+from typing import Dict
+
+
+_JSON_KEYS = {
+    type(None): 'nullValue',
+    bool: 'booleanValue',
+    int: 'integerValue',
+    float: 'doubleValue',
+    # TODO timestampValue
+    # TODO keyValue
+    str: 'stringValue',
+    # TODO blobValue
+    # TODO geoPointValue
+    # TODO entityValue
+    # TODO arrayValue
+}
+
+_VALUE_TYPES = {v: k for k, v in _JSON_KEYS.items()}
+
+
+# https://cloud.google.com/datastore/docs/reference/data/rest/v1/projects/runQuery#value
+class Value:
+    def __init__(self, value: Any, exclude_from_indexes: bool = False):
+        self.value = value
+        self.excludeFromIndexes = exclude_from_indexes
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Value):
+            return False
+        return bool(
+            self.excludeFromIndexes == other.excludeFromIndexes
+            and self.value == other.value)
+
+    def __repr__(self) -> str:
+        return str(self.to_repr())
+
+    @classmethod
+    def from_repr(cls, data: Dict[str, Any]) -> 'Value':
+        for json_key, value_type in _VALUE_TYPES.items():
+            if json_key in data:
+                if json_key == 'nullValue':
+                    value = None
+                else:
+                    value = value_type(data[json_key])
+                break
+        else:
+            raise NotImplementedError(
+                f"Dict '{str(data)}' does not contain a supported value key")
```

Comment:
It'd be great to be a bit more specific here -- rather than serializing the entire dictionary, we could just print out the actual type which is failing, eg, something like
```
raise NotImplementedError(f'{data.keys()} is not a supported value type')
```

FYI: No need to call `str` within an f-string, that'll be done automatically.

Reply:
The problem for this specific error message is that we also end up here if the type is missing altogether. That's why I print the whole dict. But I guess printing the dict is not that useful without also printing the list of supported values.

The best way to handle errors is probably to
1.  Count the number of fields and raise an error if it's different than 2, then
2. Validate the field that is not called 'excludeFromIndexes'

There is also a 'meaning' field that Google says 'should only be populated for backwards compatibility', and I'm not sure if it's actually required - still need to experiment when I can run end-to-end tests. But that means I may not know how many fields are expected (could be 2 or 3)

Reply:
That might just just point to a difference in the error message: do you think raising
```
raise NotImplementedError(f'{data.keys()} does not contain a supported value type (any of: {_VALUE_TYPES.keys()})')
```
would do the trick?

Reply:
Yes, I think this is the best. The error message is not that precise but it does give all the information needed